src/test/fuzz/minisketch.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <minisketch.h>
+#include <minisketchwrapper.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <util/check.h>
+
+#include <map>
+#include <numeric>
+
+FUZZ_TARGET(minisketch)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    const auto capacity{fuzzed_data_provider.ConsumeIntegralInRange<size_t>(1, 200)};
+    Minisketch sketch_a{Assert(MakeMinisketch32(capacity))};
+    Minisketch sketch_b{Assert(MakeMinisketch32(capacity))};
+
+    // Fill two sets and keep the difference in a map
+    std::map<uint32_t, bool> diff;
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 10000)
+    {
+        const auto entry{fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(1, std::numeric_limits<uint32_t>::max() - 1)};
+        const auto KeepDiff{[&] {
```

Comment:
> Am I correct?

Yes

Reply:
Ok, then my concern is: collisions like this should be so rare that I'm not even sure we should have the logic to catch these cases (maybe just a comment would be good enough)?

But, if we do want to account for these collisions correctly, then I'm not sure this does the job. Bc/ in this case if you add the same element to `sketch_a` twice, it looks like the diff gets flipped. etc...

All things considered, I'm wondering if it will be most robust to have a `set_a` and `set_b` alongside the sketches, `num_differences`) is then just the size of that set after the `LIMITED_WHILE`

Reply:
Adding the same element twice to a minisketch set will erase it. This is a feature, not a rare edge case.

Reply:
Ah wow, interesting. I kinda wondered this while reading the README but figured it wasn't the case. Anyways, would it make sense to reduce the max possible value of `const auto entry{fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(1, std::numeric_limits<uint32_t>::max() - 1)};` so that we see these "collisions" normally and get more test coverage?

Reply:
I expect the fuzz engine to copy the bytes in the input, so a collision might even be the "normal" case.

Reply:
What @MarcoFalke said, but also: "rare" isn't relevant for fuzz tests. The inputs are uniformly distributed; the fuzzer is actively trying to bias the inputs to trigger "rare" edge cases.

Reply:
Interesting, thanks